dali/imgcodec/parsers/webp.cc

```diff
@@ -12,19 +12,118 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <string>
 #include "dali/imgcodec/parsers/webp.h"
+#include "dali/core/byte_io.h"
 
 namespace dali {
 namespace imgcodec {
 
+namespace {
+  bool is_pattern_matching(const uint8_t *data, const std::string &pattern) {
```

Comment:
We can consider:
```suggestion
  template <size_t N>
  bool is_pattern_matching(const std::array<uint8_t, N> &data, const char(&pattern)[N + 1]) { 
```
(N +  1) for the `\0` terminator, we can get the string literal directly and have it match the size of the array.

Reply:
why `(&pattern)[N + 1]` instead of plain `pattern[N + 1]`?